projects/knora-ui/src/lib/viewer/values/color-value/color-value.component.ts

```diff
@@ -0,0 +1,124 @@
+import { Component, Inject, Input, OnChanges, OnDestroy, OnInit, SimpleChanges, ViewChild } from '@angular/core';
+import { FormBuilder, FormControl, FormGroup, FormGroupDirective, NgForm, ValidatorFn, AbstractControl } from '@angular/forms';
+import { CreateColorValue, ReadColorValue, UpdateColorValue } from '@knora/api';
+import { Subscription } from 'rxjs';
+import { BaseValueComponent } from '../base-value.component';
+import { ColorPickerComponent, ColorPicker } from './color-picker/color-picker.component';
+import { ErrorStateMatcher } from '@angular/material';
+import { CustomRegex } from '../custom-regex';
+
+/** Error when invalid control is dirty, touched, or submitted. */
+export class ColorErrorStateMatcher implements ErrorStateMatcher {
+  isErrorState(control: FormControl | null, form: FormGroupDirective | NgForm | null): boolean {
+    const isSubmitted = form && form.submitted;
+    return !!(control && control.invalid && (control.dirty || control.touched || isSubmitted));
+  }
+}
+
+@Component({
+  selector: 'kui-color-value',
+  templateUrl: './color-value.component.html',
+  styleUrls: ['./color-value.component.scss']
+})
+export class ColorValueComponent extends BaseValueComponent implements OnInit, OnChanges, OnDestroy {
+
+  @ViewChild('colorInput', { static: false }) colorPickerComponent: ColorPickerComponent;
+
+  @Input() displayValue?: ReadColorValue;
+
+  valueFormControl: FormControl;
+  commentFormControl: FormControl;
+  form: FormGroup;
+  valueChangesSubscription: Subscription;
+  customValidators = [];
+  matcher = new ColorErrorStateMatcher();
+
+  constructor(@Inject(FormBuilder) private fb: FormBuilder) {
+    super();
+  }
+
+  standardValidatorFunc: (val: any, comment: string, commentCtrl: FormControl) => ValidatorFn
```

Comment:
I think if you make the value a string (instead of an instance of `ColorPicker`), then you do not have to override the base class's `standardValidatorFunc`. Instead, you can just provide `customValidators = [Validators.pattern(CustomRegex.COLOR_REGEX)];`